src/lib/signature_lib/schnorr.ml

```diff
@@ -210,17 +220,37 @@ module Make
 
   let is_even (t : Field.t) = not (Bigint.test_bit (Bigint.of_field t) 0)
 
-  let sign (d_prime : Private_key.t) m =
+  let sign ?signature_kind (d_prime : Private_key.t) (m : Message.t) =
```

Comment:
why not add a "chain_id" to data structures like transactions instead? This is what other cryptocurrencies do.